### ISSUE TITLE
False-positive unresolved reference error message

### ISSUE THREAD

**Issue by Undin** [intellij-rust/intellij-rust#5965]
## Environment

* **IntelliJ Rust plugin version:** 0.3.129.3287-202-nightly
* **Rust toolchain version:** 1.45.2 (d3fb005a3 2020-07-31) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2020.2.1 Preview Ultimate Edition (IU-202.6948.5)
* **Operating system:** macOS 10.13.6

## Problem description
If you call trait method via UFCS, trait name is not in the scope and it's used via a qualified path, the plugin shows false-positive unresolved reference error and suggests to import the corresponding trait.
![image](https://user-images.githubusercontent.com/2539310/90689143-3caac980-e278-11ea-9ed4-6cf27eee7dfb.png)

## Steps to reproduce
```rust
mod foo {
    pub trait Bar {
        fn baz(&self) {}
    }
}
struct S;
impl foo::Bar for S {}
fn main() {
    foo::Bar::baz(&S)
}
```
